pkg/cstor/cstor.go

```diff
@@ -534,16 +534,20 @@ func (p *Plugin) SetVolumeID(unstructuredPV runtime.Unstructured, volumeID strin
 	vol := p.volumes[volumeID]
 
 	if p.local {
-		fsType := pv.Spec.PersistentVolumeSource.ISCSI.FSType
-
-		pv.Spec.PersistentVolumeSource = v1.PersistentVolumeSource{
-			ISCSI: &vol.iscsi,
+		if !vol.isCSIVolume {
+			fsType := pv.Spec.PersistentVolumeSource.ISCSI.FSType
+			pv.Spec.PersistentVolumeSource = v1.PersistentVolumeSource{
+				ISCSI: &vol.iscsi,
+			}
+			// Set Old PV fsType
+			pv.Spec.PersistentVolumeSource.ISCSI.FSType = fsType
+		} else {
+			fsType := pv.Spec.PersistentVolumeSource.CSI.FSType
+			pv.Spec.PersistentVolumeSource.CSI.VolumeHandle = vol.volname
+			// Set Old PV fsType
+			pv.Spec.PersistentVolumeSource.CSI.FSType = fsType
```

Comment:
setting `fsType` not required since we are not replacing `PersistentVolumeSource`

Reply:
Got it